### PR TITLE
Use current recommended start options for http-kit thus allowing dynamic port selection

### PR DESCRIPTION
**Pull request by cch1**
When running automated tests, it is convenient to select a socket dynamically when starting the server.  This PR introduces support for the convention of port 0 as a request for a dynamic port.

This PR required some minor changes to logging and I've also updated the version of timbre to the latest.

### CODE REVIEW

File: src/braid/core/server/core.clj [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
-    [braid.core.server.handler :refer [mobile-client-app
-                                       desktop-client-app
-                                       api-server-app]]
-    [braid.base.server.ws-handler] ; for mount
-    [braid.core.server.sync] ; for multimethods
-    [braid.base.server.jobs] ; for mount
-    [mount.core :as mount :refer [defstate]]
-    [org.httpkit.server :refer [run-server]]
-    [taoensso.timbre :as timbre]))
+  (:require [braid.core.server.handler :refer [mobile-client-app
```
Comment (jamesnvc) [inline src/braid/core/server/core.clj:2]: I think we're generally trying to move the `ns` forms to the `:require`-on-its-only line style; could you change this back to be like

```
(:require
   [braid.core.server.handler :refer [mobile-client-app]]
   ....)
```

Comment (cch1) [reply src/braid/core/server/core.clj:2]: I can make that change and I appreciate the emphasis on consistency.  

I note that this approach is uncommon -pick a random dependency of braid and observe the approach used in its `ns` forms.  More frustrating: the indenting you exhibit in your example is not [standard](https://guide.clojure.style/#one-space-indent) nor even in the [common exceptions](https://stuartsierra.com/2016/clojure-how-to-ns.html) to the standard and will require custom emacs Lisp configuration to accommodate.

Your codebase, your rules.  Please let me know your decision.

Comment (jamesnvc) [reply src/braid/core/server/core.clj:2]: Oh sorry, the indentation I'm fine to change -- I'm also using Emacs now, so I'm happy to go with it does by default :grin:. I believe the Stuart Sierra "How to NS" is the style we're trying to migrate to.

Comment (cch1) [reply src/braid/core/server/core.clj]: Done!

**Comment by jamesnvc** [comment]
Awesome, thanks!

**Comment by jamesnvc** [comment]
I merged this in, but it looks like it has an issue with config now having incorrect values. I will try to determine a convenient fix...

**Comment by jamesnvc** [comment]
Some slight annoyances to deal with circular dependencies, but I think I have something that will work now

**Comment by jamesnvc** [comment]
So, the issue is that when setting the port as zero, config picks that up too, so the default value for `:api-domain` and `:site-url` are incorrect (i.e. based on a port of zero, not whatever was automatically found). Simply requiring server into conf doesn't work properly, since that introduces a circular dependency. Tried adding an extra atom that the server would swap the values into on start, but that happens after conf is "started" and so doesn't get picked up.

If a particular api-domain & site-url are set, then this isn't a problem, but it does make testing with localhost annoying, so I will continue to try to find a reasonable workaround.

**Comment by cch1** [comment]
Randomized ports are only viable for automated tests and maybe development.  In the short term, I would consider it the developer's responsibility to know the limitations you have identified.

In the long term, IMHO, both `:api-domain` and `:site-url` should be derived at runtime based on the request.  Either the browser-supplied attributes present in the standard ring request or their overrides (`x-forwarded-...`/`Fowarded` headers) in virtualized environments allow for that.  The one minor exception being batch/async processing outside the context of a request -and in that case those values probably need to be set explicitly anyways.

**Comment by jamesnvc** [comment]
> Randomized ports are only viable for automated tests and maybe development. In the short term, I would consider it the developer's responsibility to know the limitations you have identified.

Agreed, but I want it to at least work properly in dev; without 7250969, it appears to load, but logging in fails in a very non-obvious way. Now it works, albeit as kind of a hack, but I think that's okay since, as you say, it's only relevant in test & dev.

> In the long term, IMHO, both :api-domain and :site-url should be derived at runtime based on the request

I don't know if that will be possible for `:site-url`, since that's mainly used in the backend to send emails and such, so not necessarily tied to a particular request.

`:api-domain` is needed for the front end to know where to send the XHR requests to, since currently the front and api are running on separate servers. We probably could remove this distinction, since it adds complication around cookies, but on the other hand, this does make having multiple frontends a bit simpler.

**Comment by cch1** [comment]
Agreed that for the batch stuff (like email digest) random ports is an issue -but I really can't imagine a scenario where you would use random ports *and* run batch jobs *and* be in anything close to a production scenario.  And in development, caveat developer.

Another hack (versus unifying the ws and http servers) would be to have a promise that is stored in state that holds the `:site-url` and .  On the first request where the host/x-forwarded headers are not IP-based, deliver the promise.   In any case I would avoid putting these hacks into the src tree since they probably only are relevant for devs -so maybe `dev-src`?

**Pull request merged**